packages/flutter/lib/src/material/combo_box.dart

```diff
@@ -0,0 +1,747 @@
+Map<ShortcutActivator, Intent> _kMenuTraversalShortcuts = <ShortcutActivator, Intent> {
+  LogicalKeySet(LogicalKeyboardKey.arrowUp): const _ArrowUpIntent(),
+  LogicalKeySet(LogicalKeyboardKey.arrowDown): const _ArrowDownIntent(),
+};
+
+const double _kMinimumWidth = 112.0;
+
+const double _kDefaultHorizontalPadding = 12.0;
+
+/// Defines a [ComboBox] menu button that represents one item view in the menu.
+///
+/// See also:
+///
+/// * [ComboBox]
+class ComboBoxEntry {
+  /// Creates an entry that is used with [ComboBox.comboBoxEntries].
+  ///
+  /// [label] must be non-null.
+  const ComboBoxEntry({
+    required this.label,
+    this.leadingIcon,
+    this.trailingIcon,
+    this.enabled = true,
+    this.style,
+  });
+
+  /// The label displayed in the center of the menu item.
+  final String label;
+
+  /// An optional icon to display before the label.
+  final Widget? leadingIcon;
+
+  /// An optional icon to display after the label.
+  final Widget? trailingIcon;
+
+  /// Whether the menu item is enabled or disabled.
+  ///
+  /// The default value is true. If true, the [ComboBoxEntry.label] will be filled
+  /// out in the text field of the [ComboBox] when this entry is clicked; otherwise,
+  /// this entry is disabled.
+  final bool enabled;
+
+  /// Customizes this button's appearance.
+  ///
+  /// Null by default.
+  final ButtonStyle? style;
+}
+
+
+/// A dropdown menu that can be opened from a [TextField]. The selected
+/// menu item is displayed in that field.
+///
+/// This widget is used to help people make a choice from a menu and put the
+/// selected item into the text input field. People can also filter the list based
+/// on the text input or search one item in the menu list.
+///
+/// The menu is composed of a list of [ComboBoxEntry]s. People can provide information,
+/// such as: label, leading icon or trailing icon for each entry. The [TextField]
+/// will be updated based on the selection from the menu entries. The text field
+/// will stay empty if the selected entry is disabled.
+///
+/// The dropdown menu can be traversed by pressing the up or down key. During the
+/// process, the corresponding item will be highlighted and the text field will be updated.
+/// Disabled items will be skipped during traversal.
+///
+/// The menu can be scrollable if not all items in the list are displayed at once.
+///
+/// {@tool dartpad}
+/// This sample shows how to display outlined [ComboBox] and filled [ComboBox].
+///
+/// ** See code in examples/api/lib/material/combo_box/combo_box.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+/// * [MenuAnchor], which is a widget used to mark the "anchor" for a set of submenus.
+///   The [ComboBox] uses a [TextField] as the "anchor".
+/// * [TextField], which is a text input widget that uses an [InputDecoration].
+/// * [ComboBoxEntry], which is used to build the [MenuItemButton] in the [ComboBox] list.
+class ComboBox extends StatefulWidget {
```

Comment:
+1, I will note "Dropdown Menu" is the recommended term from Material. See https://docs.google.com/spreadsheets/d/1R1YFJhEmiKIe_c9Sq1AVDOAMdTf57RvaQwoBZ_8zDHg/edit#gid=0&range=D157

Reply:
Makes sense to me:) Just for more information, I also saw some documentation here: https://m3.material.io/components/menus/guidelines#73531839-5ef8-46cd-9a5e-223f32912edb

But, I'm okay with both. DropDownMenuBox is definitely more understandable.

Reply:
Fair enough. I don't have a strong opinion here, but my guess is that most people looking for a widget like this would not think to look for it by 'ComboBox'. At least with 'DropdownMenuBox' it is closer to the main term used in the spec. To me just 'DropdownMenu' implies that it is just the menu and not the text field driving it. But like I said, I don't feel that strongly about it, and am open to other suggestions or just leaving it as 'ComboBox'.

Reply:
As @QuncCccccc pointed out, no less than the M3 spec says:

> A menu with filtering is also known as autocomplete or a _combobox_.

 I think DropdownMenuBox is the most accurate option however this widget is likely to be heavily used and a clunky name doesn't become less so the more it is used. Maybe we could just use DropdownMenu with the rationale that "dropdown" refers to both the box and the way the menu appears.